### PR TITLE
Threshold for Dice loss

### PR DESCRIPTION
**Pull request by abdumhmd**
Added a threshold parameter for Dice Loss.
    `def __init__(self, eps=1.0, beta=1.0, activation=None, threshold=None, ignore_channels=None, **kwargs):`

After doing so, the `_threshold` function in `utils/functional.py` causes the following error
`element 0 of tensors does not require grad and does not have a grad_fn loss function`

So, fixed that by making the function return a tensor that requires grad as follows
`torch.tensor((x > threshold).type(x.dtype),requires_grad=True)`

### CODE REVIEW

**Comment by qubvel** [comment]
Sorry utils module is stale and will not be updated
